packages/core/src/modules/event/event-repository.ts

```diff
@@ -38,6 +39,26 @@ export class EventRepositoryImpl implements EventRepository {
     return events.map((e) => mapToEvent(e))
   }
 
+  async getAllByUserAttending(userId: string): Promise<Event[]> {
+
+    const event_ids = await this.db
+      .selectFrom("attendance")
+      .leftJoin("attendee", "attendee.attendanceId", "attendance.id")
+      .select("attendance.eventId")
+      .where("attendee.userId", "=", userId)
+      .groupBy("attendance.eventId")
+      .execute()
+
+    const eventPromises = event_ids
+      .map(({ eventId }) => eventId)
+      .filter((id): id is string => id !== null)
+      .map(async (id) => this.getById(id))
```

Comment:
Is it possible to change this to a left join onto events? I think the current code causes an 1+N query

Reply:
it should now use 2 queries